presto-parquet/src/main/java/com/facebook/presto/parquet/writer/ParquetWriter.java

```diff
@@ -259,11 +262,21 @@ static Slice getFooter(List<RowGroup> rowGroups, MessageType messageType)
     {
         FileMetaData fileMetaData = new FileMetaData();
         fileMetaData.setVersion(1);
-        fileMetaData.setSchema(MessageTypeConverter.toParquetSchema(messageType));
         long totalRows = rowGroups.stream().mapToLong(RowGroup::getNum_rows).sum();
         fileMetaData.setNum_rows(totalRows);
         fileMetaData.setRow_groups(ImmutableList.copyOf(rowGroups));
 
+        // use parquet metadata converter to get schema with proper logical type info
+        org.apache.parquet.hadoop.metadata.FileMetaData parquetMetaDataInput = new org.apache.parquet.hadoop.metadata.FileMetaData(
+                messageType,
+                ImmutableMap.of(),
+                fileMetaData.getCreated_by());
+
+        ParquetMetadataConverter metadataConverter = new ParquetMetadataConverter();
+        FileMetaData parquetMetaData = metadataConverter.toParquetMetadata(1, new ParquetMetadata(parquetMetaDataInput, ImmutableList.of()));
```

Comment:
I believe this method is stateless.  If so, can you make `metadataConverter` a constant?